crates/bdk/src/wallet/mod.rs

```diff
@@ -969,20 +991,8 @@ impl<D> Wallet<D> {
         // sort input/outputs according to the chosen algorithm
         params.ordering.sort_tx(&mut tx);
 
-        let txid = tx.txid();
-        let sent = coin_selection.local_selected_amount();
         let psbt = self.complete_transaction(tx, coin_selection.selected, params)?;
-
-        let transaction_details = TransactionDetails {
-            transaction: None,
-            txid,
-            confirmation_time: ConfirmationTime::Unconfirmed { last_seen: 0 },
-            received,
-            sent,
-            fee: Some(fee_amount),
```

Comment:
I am concerned that we can't reproduce this `fee` value unless we return it here. When you use `add_foreign_utxo` to the builder you are telling it the value of the input. This won't exist in the `TxGraph` (although it could we have a `mut self` here!).

So we can:
1) add back the second return value to return the fee
2) Keep only one return value but insert the `TxOut`s into the `TxGraph` so we can later query for fee and get the right answer (they would be un-anchored  i.e. assumed to be in the mempool). Thse txouts would be committed to disk so we could query the fee of this tx later on.
4) Require the user inserts the foreign utxo into the wallet manually before they can use it in `add_foreign_utxo`.
5) Simply recommend that the user always insert the foreign UTXO into the wallet manually if they want to be able to calculate the fees of the resulting transaction. This makes the user think about whether they can actually trust the values they have in this UTXO.

I think I lean towards (5) because we have planning module integration coming up and that may change transaction building a lot. I notice that in order to do (4) or (5) we'd need to add `insert_txout` method on wallet that calls the graph's method.

Reply:
Interesting problem. I agree the last option (4) makes sense, I can add some documentation to explain the situation. I'll also add a test to reproduce the situation to confirm it works as expected.

Reply:
The other option will change the `TxGraph::calculate_fee` method to include external outputs.

```rust
pub fn calculate_fee(
    &self, 
    tx: &Transactions, 
    missing_outputs: impl IntoIterator<Item = (OutPoint, TxOut)>,
) -> Result<i64, CalculateFeeError> {
    unimplemented!()
}
```

Reply:
In a future `1.x` release we could add a `calculate_fee_with_outputs` function is there's any demand for it.

Reply:
To test the wallet calculate_fee  and fee_rate with and without foreign utxos I added `Wallet::insert_txout`, which exposes the underlying TxGraph::insert_txout. For this new function I'm also persisting the change, as long as the txout data is trusted this is OK isn't it? 

I also updated docs with warnings about missing utxos, and to trust any added. Is it clear enough?